test/functional/tests/lazy_writes/test_lazy_writes_clean.py

```diff
@@ -0,0 +1,259 @@
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(512, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+        Udev.disable()
+
+    with TestRun.step(f"Start cache in {cache_mode} mode."):
+        cache = casadm.start_cache(cache_part, cache_mode)
+
+    with TestRun.step("Add core to cache."):
+        core_part.create_filesystem(Filesystem.xfs)
+        core = cache.add_core(core_part)
+        core.mount(mnt_point)
+
+    with TestRun.step("Disable cleaning and sequential cutoff policies."):
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+        # get IO stats independent from test
+        exp_obj_stats_diff = IOstat.get_iostat(core).total_writes.value
+        core_disk_writes_diff = IOstat.get_iostat(core_part).total_writes.value
+
+    with TestRun.step("Write data to exported object."):
+        test_file_main = create_random_test_file("/tmp/test_file_main", Size(64, Unit.MebiByte))
+        dd = Dd().output(mnt_point + "/test_file") \
+            .input(test_file_main.full_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_main.refresh_item()
+        test_file_md5sum_main = test_file_main.md5sum()
+
+    with TestRun.step("Check IO stats."):
+        core_disk_writes = IOstat.get_iostat(core_part).total_writes.value - core_disk_writes_diff
+        exp_obj_stats = IOstat.get_iostat(core).total_writes.value - exp_obj_stats_diff
+        if core_disk_writes > 0:
+            TestRun.LOGGER.error("Writes should occur only on cache device.")
+        if exp_obj_stats != test_file_main.size.value:
+            TestRun.LOGGER.error("Not all writes reached exported object.")
+
+    with TestRun.step("Read data from exported object."):
+        test_file_1 = File.create_file("/tmp/test_file_1")
+        dd = Dd().output("/tmp/test_file_1") \
+            .input(mnt_point + "/test_file") \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_1.refresh_item()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if test_file_md5sum_main != test_file_1.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be equal.")
+
+    with TestRun.step("Read data from core device."):
+        test_file_2 = File.create_file("/tmp/test_file_2")
+        dd = Dd().output("/tmp/test_file_2") \
+            .input(core_part.system_path) \
+            .block_size(bs) \
+            .count(int(test_file_main.size / bs)) \
+            .oflag("direct")
+        dd.run()
+        test_file_2.refresh_item()
+
+    with TestRun.step("Compare md5 sum of test files."):
+        if test_file_md5sum_main == test_file_2.md5sum():
+            TestRun.LOGGER.error("Md5 sums should be different.")
```

Comment:
You are reading raw data from device with filesystem, so this check doesn't have to much sense.